src/lib/structures/expr.ml

```diff
@@ -1230,7 +1226,7 @@ let is_skolem_cst v =
 
 let get_skolem =
   let hsko = Hsko.create 17 in
-  let gen_sko ty = mk_term (Sy.fresh "@sko") [] ty in
+  let gen_sko ty = mk_term (Sy.fresh_skolem "@sko") [] ty in
```

Comment:
I think that `Sy.fresh` used to add an extra `__` prefix (with the comment "(* garder le suffixe "__" car cela influence l'ordre *)") but it looks like `Sy.fresh_skolem` doesn't, so unless I missed something we used to get `!?__@sko` and we now get `!?@sko` here.

Reply:
> I think that `Sy.fresh` used to add an extra `__` prefix (with the comment "(* garder le suffixe "__" car cela influence l'ordre *)") but it looks like `Sy.fresh_skolem` doesn't, so unless I missed something we used to get `!?__@sko` and we now get `!?@sko` here.

That makes me think this whole check system is unsound, because an identifier can start with `!?` if is has been defined quoted.

Reply:
This is why we want to replace `!` with `.`! Identifiers starting with `.` or `@` are reserved for solver usage and cannot be used in scripts, dolmen ensures this (or will in the next release, see Gbury/dolmen#185 ).

Reply:
> Ah, sorry, I misread your remark — indeed, according to the SMT-LIB it can, but I think after [Gbury/dolmen#185](https://github.com/Gbury/dolmen/issues/185) identifiers starting with `.?` are rejected even in quoted form (which is incorrect from a strict reading of the standard).

You did not, I miswrote it and edited, you might have seen the old version :)